react/features/transcription/middleware.js

```diff
@@ -0,0 +1,40 @@
+import { JitsiConferenceEvents } from '../base/lib-jitsi-meet';
+
+import {
+    CONFERENCE_FAILED,
+    CONFERENCE_JOINED,
+    CONFERENCE_LEFT
+} from '../base/conference';
+import { MiddlewareRegistry } from '../base/redux';
+
+import { updateTranscriptionState } from './actions';
+
+/**
+ * Middleware that captures conference actions and adds a listener to
+ * transcription status events.
+ *
+ * @param {Store} store - Redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+
+    switch (action.type) {
+    case CONFERENCE_JOINED: {
+        const { conference } = action;
+
+        conference && conference.on(
+            JitsiConferenceEvents.TRANSCRIPTION_STATUS_CHANGED,
+            (...args) => store.dispatch(updateTranscriptionState(...args)));
+
+        break;
+    }
+
+    case CONFERENCE_FAILED:
+    case CONFERENCE_LEFT:
+        // REMOVE THE LISTENER?
```

Comment:
Maybe for safety. The rule I follow is every .on should have a corresponding .off. It's not really a problem for the web app though because a page refresh is done on conference leave.

Reply:
Yeah, I have the same thinking and then I searched for off-s in our code and didn't find any and decided to ask about it.